### PR TITLE
Bugfix v2.0.2

### PR DESCRIPTION
**Pull request by fsimonis**
## How to work with this template

* [x] assign a release manager, who takes care of the process
* [ ] assign each point below to a responsible person, before you continue. Use `@member`.

Only the release manager should update this post (even tickboxes, due to race conditions in editing). Everybody else should comment on the PR with the progress.

## Step by step guide
* [x] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) (all)
   * Add things, if necessary
   * Extract summary
   * Fix wording
   * Sort the entries lexicographically
* [x] Look over the Roadmap and update entries.
* [x] Merge master to develop (No commits after the release on master)
* [x] Check code base w.r.t code formatting (run [`precice/tools/formatting/check-format`](https://github.com/precice/precice/blob/develop/tools/formatting/check-format)) and reformat if required (run [`precice/tools/formatting/format-all`](https://github.com/precice/precice/blob/develop/tools/formatting/format-all))
* [x] Create branch `release-N` from develop. If needed, `git rebase develop`.
* [x] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md))
* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
* [x] Fix potential problems in develop (all)
* [x] Run `tools/releasing/bumpversion.sh MAJOR.MINOR.PATCH` to bump the version
* [x] Verify the version changes in:
   * [x] [CHANGELOG](https://github.com/precice/precice/blob/develop/CHANGELOG.md)
   * [x] [CMakeLists.txt](https://github.com/precice/precice/blob/develop/CMakeLists.txt)
   * [x] [debian changelog](https://github.com/precice/precice/blob/develop/tools/releasing/packaging/debian/changelog)
   * [ ] (do independent release, if necessary!) [Python bindings](https://github.com/precice/python-bindings)
   * [ ] (do independent release, if necessary!) [MATLAB bindings](https://github.com/precice/matlab-bindings)
* [ ] Draft message to mailing list
* [ ] Update documentation (all)
  * [ ] Update markdown configuration reference in wiki
* [ ] Approve the PR with at least two reviews (all)
* [ ] Merge PR to master 
* [ ] Tag release on master and verify by running `git describe --tags` (on GitHub, make sure to select the release branch as target)
* [ ] Merge back to develop and verify by running `git describe --tags`

## Regression Tests

Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289

* [ ] SU2 / CalculiX [flap_perp](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp/SU2-CalculiX)
* [ ] OpenFOAM / OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)
* [ ] OpenFOAM / OpenFOAM - NP mapping in OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)
* [ ] OpenFOAM / CalculiX FSI [flap perp](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp/OpenFOAM-CalculiX)
* [ ] OpenFOAM / CalculiX FSI - NP mapping in CalculiX [3D_Tube](https://github.com/precice/tutorials/tree/develop/FSI/3D_Tube/OpenFOAM-CalculiX)
* [ ] OpenFOAM / CalculiX / OpenFOAM CHT [heat_exchanger](https://github.com/precice/tutorials/tree/develop/CHT/heat_exchanger/buoyantSimpleFoam-CalculiX)
* [ ] OpenFOAM / deal.II [flap_perp_2D](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp_2D/OpenFOAM-deal.II)
* [ ] OpenFOAM / FEniCS [flap_perp](https://github.com/precice/tutorials/tree/master/FSI/flap_perp/OpenFOAM-FEniCS)
* [ ] OpenFOAM / FEniCS [flow-over-plate](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-fenics)
* [ ] OpenFOAM / Nutils [flow-over-plate](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-nutils)
* [ ] FEniCS / FEniCS [partitioned-heat](https://github.com/precice/tutorials/tree/master/HT/partitioned-heat/fenics-fenics)
* [ ] MATLAB / MATLAB [ODEs](https://github.com/precice/matlab-bindings/tree/develop/tutorial)
* [ ] ExaFSA: Ateles / FASTEST
* [x] Alya
* [ ] 1D-ElasticTube [C++](https://github.com/precice/elastictube1d/tree/develop/cxx)
* [ ] 1D-ElasticTube [Python](https://github.com/precice/elastictube1d/tree/develop/python)
* [ ] SuperMUC
* [x] Solverdummy [C++](https://github.com/precice/precice/tree/develop/tools/solverdummies/cpp)
* [x] Solverdummy [C](https://github.com/precice/precice/tree/develop/tools/solverdummies/c)
* [x] Solverdummy [Fortran](https://github.com/precice/precice/tree/develop/tools/solverdummies/fortran)
* [ ] Solverdummy [Fortran 2003](https://github.com/precice/precice/tree/develop/tools/solverdummies/f2003)
* [ ] Solverdummy [Python](https://github.com/precice/python-bindings/tree/develop/solverdummy)
* [ ] Solverdummy [MATLAB](https://github.com/precice/matlab-bindings/tree/develop/solverdummy)

## Post-release
* [ ] Generate packages
   * [ ] Latest Ubuntu LTS
   * [ ] Latest Ubuntu
   * [ ] Arch Linux AUR Package
* [ ] Update Spack recipe
* [ ] Send email and do marketing
* [ ] Tweet
* [ ] Update the [PR template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)

To open a new PR with this template, use this [PR template query](https://github.com/precice/precice/compare/new?template=release_pull_request_template.md)

### CODE REVIEW

**Comment by uekerman** [comment]
* Unit and integration tests work for me except the expected MPI Ports (OpenMPI) failures.
* I used the Alya Flap testcase to test all settings of `geometric-filter` -> all good.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 All notable changes to this project will be documented in this file. For future plans, see our [Roadmap](https://github.com/precice/precice/wiki/Roadmap).
 
+## 2.0.2
+
+- Fixed a critical bug in the testing framework.
+- Fixed a critical bug in the partitioning with two-level initialization enabled.
```
Comment (uekerman) [inline CHANGELOG.md:8]:
```suggestion
- Fixed a critical bug in the partitioning for geometric filtering on master.
```
This also affects partitioning when two-level init is off.

Comment (fsimonis) [reply CHANGELOG.md:8]: We should probably also mention that this is a setting in the use-mesh tag and that this bug does not affect the default.

```suggestion
- Fixed a critical bug in the partitioning for geometric filter set to `on-master` in `<use-mesh>` tags. The default configuration is `on-slave`.
```

Comment (uekerman) [reply CHANGELOG.md:8]: I agree

**Pull request merged**